src/charm.py

```diff
@@ -176,22 +180,35 @@ def _set_alerts(self, container):
                 alert rule files need to be created.
         """
         if not self.provider_ready:
+            logger.debug("Abort processing alert rules provider not ready!")
             return
 
-        with container.is_ready():
-            container.remove_path(RULES_DIR, recursive=True)
+        logger.debug("Processing alert rules")
 
-        for rel_id, alerts in self.prometheus_provider.alerts().items():
-            filename = "juju_{}_{}_{}_rel_{}_alert.rules".format(
-                alerts["model"], alerts["model_uuid"], alerts["application"], rel_id
-            )
-            try:
-                path = os.path.join(RULES_DIR, filename)
-                rules = yaml.dump({"groups": alerts["groups"]})
-                logger.debug("Rules for relation %s : %s", rel_id, rules)
-                container.push(path, rules, make_dirs=True)
-            except ConnectionError:
-                logger.error("Failed to write alert rule %s", filename)
+        container.remove_path(RULES_DIR, recursive=True)
+
+        alert_rules_by_rel_id = self.prometheus_provider.alerts()
```

Comment:
I presume this is just a style change. If so I like it because it makes more explicit the return value that is being iterated on.